[Wu Yuxin] iP

yuxinwu99 opened the pull request.

yuxinwu99 committed:
Level 0 greetings and good bye

yuxinwu99 committed:
added echo function with addLines function

yuxinwu99 committed:
Added List function

yuxinwu99 committed:
Add Task.java

yuxinwu99 committed:
Adjust to coding standards

yuxinwu99 committed:
Add todo deadline event commands

change if block in main to case block

yuxinwu99 committed:
Automate testing

yuxinwu99 committed:
Adjust to coding standard

yuxinwu99 committed:
Update runtest

Jane-Ng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,51 @@
         System.out.println("Hello from\n" + logo);
+        addLines("Hello! I'm Duke\n" + "What can I do for you?");
+
+        while(!input.equals("bye")) {
+            input = in.nextLine();
+            String[] splitInput = input.split(" ", 2);
+            switch(splitInput[0]){
+            case "bye":
+                addLines("Bye. Hope to see you again soon!");
+                break;
+            case "list":
+                list.queryItems();
+                break;
+            case "done":
+                list.completeTask(Integer.parseInt(splitInput[1])-1);
+                break;
+            case "todo":
+                list.addItem(splitInput[1]);
+                break;
+            case "deadline":
+                list.addDeadline(splitInput[1]);
+                break;
+            case "event":
+                list.addEvent(splitInput[1]);
+                break;
+            default:
+                addLines("invalid input!");
+            }
+        }
```
Good job on your indentation. Code is organised

yuxinwu99 replied:
Thanks :D

Jane-Ng reviewed (commented):
Looks good!

Zhu-Ze-Yu commented on src/main/java/Duke.java:
```diff
@@ -1,44 +1,50 @@
+            input = in.nextLine();
+            String[] splitInput = input.split(" ", 2);
+            switch(splitInput[0]){
+            case "bye":
```
I think you can use uppercase for the case statements

yuxinwu99 replied:
I am not sure how I can convert the case statements to uppercase when the input commands given are in lower case.

Zhu-Ze-Yu commented on src/main/java/Duke.java:
```diff
@@ -1,44 +1,50 @@
+                addLines("invalid input!");
             }
         }
+    }
```
Overall the code is very clear, readable.

Zhu-Ze-Yu commented on src/main/java/List.java:
```diff
@@ -9,16 +9,10 @@
-            } else {
-                break;
-            }
+        System.out.println("____________________________________________________________");
```
Maybe you can use some methods to print the horizontal lines and other messages.

Zhu-Ze-Yu commented on src/main/java/List.java:
```diff
@@ -49,23 +42,23 @@ public void addDeadline(String input){
+
     public void addEvent(String input){
         String[] eventInputs = input.split("/at",2);
         Event newEvent= new Event(eventInputs[0].trim(),eventInputs[1].trim());
```
Good use of trim() is good, I like it.

yuxinwu99 committed:
Add method to handle errors

Modify event output from "[D]" to "[E]" and "by:" to "at:"

yuxinwu99 committed:
Merge branch 'branch-Level-5'

yuxinwu99 committed:
Add DukeException

yuxinwu99 committed:
A-Packages

yuxinwu99 committed:
Merge branch 'branch-A-Packages'

yuxinwu99 committed:
Modify throw in list

MeLoveCarbs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+                    throw new DukeException("format");
+                }
+                switch(splitInput[0]) {
+                case "bye":
+                    addLines("Bye. Hope to see you again soon!");
+                    break;
+                case "list":
+                    LIST.queryItems();
+                    break;
+                case "done":
+                    LIST.completeTask(splitInput[1]);
+                    break;
+                case "todo":
+                    LIST.addItem(splitInput[1]);
+                    break;
+                case "deadline":
+                    LIST.addDeadline(splitInput[1]);
+                    break;
+                case "event":
+                    LIST.addEvent(splitInput[1]);
```
Maybe you can replace the string literals within each case into constants to improve readability and avoid magic literals

MeLoveCarbs commented on src/main/java/duke/list/List.java:
```diff
@@ -0,0 +1,92 @@
+    public void queryItems(){
+        printLine();
+        System.out.println("Here are the tasks in your list:");
+        for(int i=0; i<size; i++) {
```
Try putting spaces in between to improve readability like so: 
`        for (int i = 0; I < size; i++) {
`

MeLoveCarbs commented on src/main/java/duke/list/List.java:
```diff
@@ -0,0 +1,92 @@
+    private int size;
+
+    public List(){
+        items = new Task[100];
```
Maybe you can try using ArrayList from java's packages so that you won't have to declare a fixed size array. What happens if user inputs more than 100 tasks?

MeLoveCarbs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+            input = IN.nextLine();
+            try {
+                String[] splitInput = input.split(" ", 2);
+                if(splitInput.length==1 && !input.equals("bye") && !input.equals("list")){
```
Maybe you can try extracting out the conditions into a method to improve readability, or give a comment to let others know what this condition does.

MeLoveCarbs commented on src/main/java/duke/list/List.java:
```diff
@@ -0,0 +1,92 @@
+        int index;
+        printLine();
+        try{
+            index=Integer.parseInt(input)-1;
```
Have you tried checking if user puts out of bounds index? For eg, 0?

MeLoveCarbs reviewed (commented):
Good job on your iP! Keep up the good effort and try to avoid using string literals. Instead, convert them to constants/enum classes that you've learnt during lecture :)

yuxinwu99 committed:
Add remove method

Modify use of array to arraylist in List.java
Modify input.txt and expected.txt

yuxinwu99 committed:
Change try catch in List.java

yuxinwu99 committed:
Add read and write list to file method

yuxinwu99 committed:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/main/java/duke/list/List.java

yuxinwu99 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/list/List.java
Modify to append when user enter command

yuxinwu99 committed:
Change key to save in duke.txt from " " to "|"

yuxinwu99 committed:
refactor to make TaskList, Ui, Parser and Storage classes

yuxinwu99 committed:
Add command classes

yuxinwu99 committed:
Fix command class name

yuxinwu99 committed:
naming bug

yuxinwu99 committed:
fix name cases

yuxinwu99 committed:
Add level 8

yuxinwu99 committed:
Fix storage reading bug

yuxinwu99 committed:
Add level 9

yuxinwu99 committed:
Add JavaDoc

yuxinwu99 committed:
Merge pull request #1 from Lezn0/branch-Level-8

Merge level 8 to master

yuxinwu99 committed:
Merge branch 'master' into branch-level-9

# Conflicts:
#	src/main/java/duke/list/TaskList.java

yuxinwu99 committed:
Merge pull request #2 from Lezn0/branch-level-9

Merge level 9 to master

yuxinwu99 committed:
Merge branch 'master' into branch-JavaDoc

# Conflicts:
#	src/main/java/duke/list/TaskList.java

yuxinwu99 committed:
Merge pull request #3 from Lezn0/branch-JavaDoc

Merge JavaDoc branch to master

yuxinwu99 committed:
Fix bug in event toString()

yuxinwu99 committed:
Merge branch 'branch-Level-8'

yuxinwu99 committed:
Add user guide

yuxinwu99 committed:
Modify user guide and help command

yuxinwu99 committed:
Set theme jekyll-theme-cayman

yuxinwu99 committed:
Add write to file after completing task

Modify help UI

yuxinwu99 committed:
Merge branch 'master' of https://github.com/Lezn0/ip

yuxinwu99 committed:
Fix due function

Add feature to state index of item in list after find and due function
